facebook/react-native
# Send Modal dismiss event on Android and remove Modal dismiss workaround

## Summary:


1. Modal `onDismiss` is currently only available on iOS. On Android, we don't have a way to know when exactly a modal is dismissed.
2. Modal `onDismiss` is not working on iOS (Fabric).

Currently, the `onDismiss` is emitted using a device event as a workaround to the `RCTModalHostView` unable to receive the event as it's already unmounted when `visible` is false. 

This PR removes the workaround and keeps `RCTModalHostView` mounted until `onDismiss` event is emitted from the host.

## Changelog:


[ANDROID] [ADDED] - Added support for Modal `onDismiss` prop
[IOS] [FIXED] - Fix `onDismiss` is not working on Fabric

## Test Plan:


1. Run rn-tester
3. Open the Modal example
4. The second example shows the counter for the show and dismiss count
5. Show and dismiss the modal and verify the count is incremented correctly

https://github.com/facebook/react-native/assets/50919443/aae354a7-b08e-476f-a1f3-a87917f29292
